include/or_ompl/RobotStateSpace.h

```diff
@@ -0,0 +1,73 @@
+#ifndef OR_OMPL_ROBOT_STATE_SPACE_H_
+#define OR_OMPL_ROBOT_STATE_SPACE_H_
+
+#include <ompl/base/spaces/RealVectorStateSpace.h>
+#include <boost/shared_ptr.hpp>
+
+namespace or_ompl {
+
+    /**
+     * Implements a state for an OpenRAVE robot
+     */
+    class RobotState : public ompl::base::RealVectorStateSpace::StateType {
+
+    public:
+        /**
+         * Constuctor - all dof values are initialized to 0
+         * @param dof_indices The dof indices this state will represent
+         */
+        RobotState(const std::vector<int> &dof_indices);
+
+        /**
+         * Clear the indices vector
+         */
+        ~RobotState();
+
+        /**
+         * Sets the values in the state. 
+         * @param dof_values The values, this ordering must match the dof_indices
+         */
+        void set(const std::vector<double> &dof_values);
+
+        /**
+         * @return The values in the state
+         */
+        std::vector<double> getValues() const;
+
+        /**
+         * @return An ordered list of indices this state corresponds to
+         */
+        std::vector<int> getIndices() const { return _indices; }
+
+    private:
+        std::vector<int> _indices;
```

Comment:
States are created via a call to allocState on the state space. The state space doesn't directly keep a handle to states its allocated. So I think technically there is no guarentee of this.  In practice, most objects in ompl (including the base planner class and the PathGeometric objects) take a SpaceInformationPtr, which has internally a shared_ptr to the StateSpace. So it is almost always true that the StateSpace would outlive the useful life of the State. Thoughts?